File: app/src/test/java/com/akhutornoy/transformerswar/interactor/battle/mars/RatingCalculatorTest.java
package com.akhutornoy.transformerswar.interactor.battle.mars;

import com.akhutornoy.transformerswar.repository.rest.dto.Transformer;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class RatingCalculatorTest {

    RatingCalculator calculator;

    @Before
    public void init() {
        calculator = new RatingCalculator();
    }

    @Test
    public void ratingShouldBe50() {
        int expected = 50;

        Transformer transformer = new Transformer.Builder()
                .setStrength(10)
                .setIntelligence(10)
                .setSpeed(10)
                .setEndurance(10)
                .setFirepower(10)
                .build();
        int actual = calculator.calculate(transformer);

        assertEquals(String.format("Rating should be %d", expected), expected, actual);
    }
}

File: app/src/test/java/com/akhutornoy/transformerswar/interactor/battle/mars/RatingCalculatorTest.kt
package com.akhutornoy.transformerswar.interactor.battle.mars

import com.akhutornoy.transformerswar.repository.cache.TransformerEntity
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test

class RatingCalculatorTest {
    internal var calculator = RatingCalculator()

    @Before
    fun init() {
        calculator = RatingCalculator()
    }

    @Test
    fun `rating Should Be 50`() {
        val expected = 50

        val transformer = TransformerEntity(
                strength = 10,
                intelligence = 10,
                speed = 10,
                endurance = 10,
                rank = 10,
                courage = 10,
                firepower = 10,
                skill = 10)
        val actual = calculator.calculate(transformer)

        assertEquals(String.format("Rating should be %d", expected), expected.toLong(), actual.toLong())
    }

}